File: java-bot/src/gos/bot/IBot.java
package gos.bot;

import gos.bot.protocol.InitiateRequest;
import gos.bot.protocol.Move;
import gos.bot.protocol.MoveRequest;
import gos.bot.protocol.ProcessedMove;

public interface IBot
{
    public void HandleInitiate(InitiateRequest request);
    public Move HandleMove(MoveRequest request);
    public void HandleProcessedMove(ProcessedMove move);
}

File: SharpBot/SharpBot/IBot.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SharpBot.Protocol;

namespace SharpBot
{
    public interface IBot
    {
        void HandleInitiate(InitiateRequest request);
        Move HandleMove(MoveRequest request);
        void HandleProcessedMove(ProcessedMove move);
    }
}
